### ISSUE TITLE
SVG images are broken on Windows installers

### ISSUE THREAD

**Issue by PhysSong** [LMMS/lmms#5810]
First reported: https://github.com/LMMS/lmms/pull/5160#issuecomment-545522520
In order to make SVG images working in LMMS from Windows installers and/or Linux AppImages, the Qt SVG library and corresponding image format plugin should be deployed. Currently, however, they are not.
- AppImage: installing the Qt SVG library is sufficient, `linuxdeployqt` will take care of deployment(no development libraries required).
- Windows: requires some CMake logic to bundle required DLLs, ~MinGW builds have an additional issues(the version of `qt5svg-mingw-w64` doesn't match with `qt5base-mingw-w64`.)~ the issue with MinGW is resolved now.

**Comment by PhysSong** [comment]
Fixed AppImages from CircleCI via LMMS/lmms-ci-docker#9, but spectrum analyzer on Windows is still an issue.

**Comment by Rossmaxx** [comment]
@DomClark is there a chance transitioning to MSVC completely might fix the version inconsistency too? 
If not, what's the best route for a fix?

**Comment by DomClark** [comment]
We use the official Qt distribution (albeit not via the official installer) for MSVC builds, so exclusively using MSVC builds would avoid the version inconsistency, yes. However, we still plan to support MinGW builds, and I'm not aware of any reason that an appropriate version of `qt5svg-mingw-w64` couldn't be built.

**Comment by Rossmaxx** [comment]
I'm moving this out of the milestone for now. If anyone feels this should be a part of 1.3, feel free to add back in.

**Comment by he29-net** [comment]
Should I perhaps look into adding a temporary workaround that replaces SVGs by bitmaps for Windows builds?

Having blurry or pixelated icons on Windows would probably be better than releasing a stable version with new spectrum analyzer full of randomly colored squares in place of a GUI. I have almost zero knowledge about our build system (let alone about Windows builds), so I can't really help with the underlying Qt SVG library issue.

**Comment by tresf** [comment]
Broken on macOS too.  I'm taking a look now.

<img width="550" alt="Image" src="https://github.com/user-attachments/assets/3dc2c9f3-be4d-4906-8a32-9b02ffb8ae8b" />

**Comment by tresf** [comment]
> Should I perhaps look into adding a temporary workaround that replaces SVGs by bitmaps for Windows builds?

Yeah, if we can't support QtSvg in the codebase, we need to remove SVGs until we do.  @Rossmaxx this needs to be re-added to 1.3

**Comment by tresf** [comment]
It's been a while since we tried to use a non-docker environment for building... Here's my attempt: https://github.com/tresf/lmms/blob/8ca5bee828166afbe79e2094137614f317c3622f/.github/workflows/build.yml.

It doesn't bomb-out on the QtSvg dependency and produces a Windows desktop installer.  I don't think it's properly bundling the QtSvg lib though.  Advice on where to look is welcome.

**Comment by tresf** [comment]
> I don't think it's properly bundling the QtSvg lib though. Advice on where to look is welcome.

Found it... 

```diff
SET(PLUGIN_FILES "")
IF(LMMS_BUILD_WIN32)
	INSTALL(FILES $<TARGET_FILE:Qt5::QWindowsIntegrationPlugin> DESTINATION platforms)
+	INSTALL(FILES $<TARGET_FILE:Qt5::Svg> DESTINATION .)
ENDIF()
```

.. sadly this doesn't fix the symptom.  I think I need to put it somewhere else.  Investigating.

**Comment by tresf** [comment]
> .. sadly this doesn't fix the symptom. I think I need to put it somewhere else. Investigating.

Got it... 

```diff
IF(LMMS_BUILD_WIN32)
        INSTALL(FILES $<TARGET_FILE:Qt5::QWindowsIntegrationPlugin> DESTINATION platforms)
+       INSTALL(FILES $<TARGET_FILE:Qt5::QSvgIconPlugin> DESTINATION iconengines)
+       INSTALL(FILES $<TARGET_FILE:Qt5::QSvgPlugin> DESTINATION imageformats)
+       INSTALL(FILES $<TARGET_FILE:Qt5::Svg> DESTINATION .)
ENDIF()
```

> the version of qt5svg-mingw-w64 doesn't match with qt5base-mingw-w64.

This situation has since improved, the libs are identical versions now.

**Comment by Rossmaxx** [comment]
Fixed in #7667
